backoffice/src/Services/ImageService.cs

```diff
@@ -0,0 +1,96 @@
+using Droits.Repositories;
+using Droits.Models.Entities;
+using Droits.Models.FormModels;
+
+namespace Droits.Services;
+
+public interface IImageService
+{
+    Task<Image> GetImageAsync(Guid id);
+    string GetRandomImageKey();
+    Task<Image> SaveImageAsync(Image image);
+    Task<Image> SaveImageFormAsync(ImageForm imageForm);
+    Task<Stream> GetImageStreamAsync(string key);
+    Task DeleteImagesForWreckMaterialAsync(Guid wmId, IEnumerable<Guid> imagesToKeep);
+}
+
+public class ImageService : IImageService
+{
+    private readonly IImageRepository _repo;
+
+
+    public ImageService(IImageRepository repo)
+    {
+        _repo = repo;
+    }
+    
+    public async Task<Image> SaveImageAsync(Image image)
+        {
+            if (image.Id == default)
+            {
+                return await AddImageAsync(image);
+            }
+            return await UpdateImageAsync(image);
+        }
+    
+    private async Task<Image> AddImageAsync(Image image)
+    {
+        return await _repo.AddAsync(image);
+    }
+
+    private async Task<Image> UpdateImageAsync(Image image)
+    {
+        return await _repo.UpdateAsync(image);
+    }
+    
+    public string GetRandomImageKey()
+    {
+        return _repo.GetRandomImage().Key;
+    }
+    public async Task<Image> GetImageAsync(Guid id) => await _repo.GetImageAsync(id);
+
+
+    public async Task<Image> SaveImageFormAsync(ImageForm imageForm)
+    {
+
+        Image image;
+        
+        if ( imageForm.Id == default )
+        {
+             
+            image = await _repo.AddAsync(
+                imageForm.ApplyChanges(new Image()));
+
+            if ( imageForm.ImageFile != null ) // Refactor to keep DRY
```

Comment:
probably just have a 

if(imageForm.ImageFile == null){
throw SomethingAboutNeedingAFile();
}

at the top of the == default block 

(if it's a new file, we definitely want an image uploaded)

Reply:
Added the if block, should I change the name of the exception to be consistent with the rest of the exceptions

Reply:
The exceptions we currently have are focused around EntityNotFound

I think** handing it here with some default exception is fine, with logging.

But as it's user input validation, it would probably make sense to have something in the controller too

if(shouldBeImage && modelDoesn'tContainImage){
AddError("You need to upload an image")
return View(nameof(Edit), form)
}

sort of thing